indico/MaKaC/conference.py

```diff
@@ -3607,91 +3593,35 @@ def canKeyModify(self):
             return False
         return session.get('modifKeys', {}).get(self.id) == modifKey
 
-    def grantModification( self, prin, sendEmail=True ):
-        email = None
-        if isinstance(prin, ConferenceChair):
-            email = prin.getEmail()
-        elif isinstance(prin, str):
-            email = prin
-        if email != None:
-            if email == "":
-                return
-            ah = AvatarHolder()
-            results=ah.match({"email":email}, exact=1)
-            #No registered user in Indico with that email
-            if len(results) == 0:
-                self.__ac.grantModificationEmail(email)
-                self.getConference().getPendingQueuesMgr().addPendingConfManager(prin, False)
-                if sendEmail and isinstance(prin, ConferenceChair):
-                    notif = pendingQueues._PendingConfManagerNotification( [prin] )
-                    mail.GenericMailer.sendAndLog(notif, self.getConference(), 'Event')
-            #The user is registered in Indico and is activated as well
-            elif len(results) == 1 and results[0] is not None and results[0].isActivated():
-                self.__ac.grantModification(results[0])
-                results[0].linkTo(self, "manager")
-        else:
-            self.__ac.grantModification( prin )
-            if isinstance(prin, AvatarUserWrapper):
-                prin.linkTo(self, "manager")
-
-    def revokeModification( self, prin ):
-        self.__ac.revokeModification( prin )
-        if isinstance(prin, AvatarUserWrapper):
-            prin.unlinkTo(self, "manager")
-
-    @unify_user_args(legacy=True)
-    def canUserModify( self, av ):
-        if av == None:
-            return False
-        if ( av == self.getCreator()) or self.getAccessController().canModify( av ):
-            return True
-        for owner in self.getOwnerList():
-            if owner.canUserModify( av ):
-                return True
-        return False
+    @unify_user_args
+    def canUserModify(self, user, role=None):
+        return self.as_event.can_manage(user, role=role)
 
-    @unify_user_args(legacy=True)
-    def canModify(self, aw_or_user):
+    def canModify(self, aw_or_user, role=None):
```

Comment:
It'd need an `isinstance(self._target, Conference)` or `hasattr(self._target, 'can_manage')` check instead of just passing the role if it's defined assuming the object handles it.

But checking for `can_manage` is probably cleaner indeed.